File: JS/baekjoon_1149.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";

const [inputN, ...input] = fs.readFileSync(filePath).toString().trim().split("\n");
const n = parseInt(inputN);

const cost = input.map((nums) => nums.split(" ").map(n => parseInt(n)))

for(let i=1; i<n; i++) {
    cost[i][0] = Math.min(cost[i-1][1]+cost[i][0], cost[i-1][2]+cost[i][0]);
    cost[i][1] = Math.min(cost[i-1][0]+cost[i][1], cost[i-1][2]+cost[i][1]);
    cost[i][2] = Math.min(cost[i-1][0]+cost[i][2], cost[i-1][1]+cost[i][2]);
}

console.log(Math.min(...cost[n-1]));

File: Python/baekjoon_1149.py
from sys import stdin as s

s = open("input.txt", "rt");

n = int(s.readline().strip())
arr = [list(map(int, s.readline().split())) for _ in range(n)]

for i in range(1, len(arr)):
    arr[i][0]= min(arr[i-1][1]+arr[i][0], arr[i-1][2]+arr[i][0])
    arr[i][1]= min(arr[i-1][0]+arr[i][1], arr[i-1][2]+arr[i][1])
    arr[i][2]= min(arr[i-1][0]+arr[i][2], arr[i-1][1]+arr[i][2])
    
print(min(arr[n-1]))